### PR TITLE
Add coverage checking to CI

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
Get code coverage running in CI

- [ ] It looks like this may be blocked on setting an env var in CI with a value from https://codecov.io/gh/wasmerio/wasmer

# Review

- [ ] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

File: .github/workflows/coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+on: [push]
+
+name: build
+
+env:
+  RUST_BACKTRACE: 1
```
Comment (nlewycky) [inline .github/workflows/coverage.yaml:6]: Is this intended to land or just for debugging this PR? Is this needed for coverage or is this for debugging when things do go wrong?

File: .github/workflows/coverage.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: 1.45.0
+          override: true
+      - name: Generate Coverage Report
+        run: |
+          cargo install cargo-tarpaulin
+          cargo tarpaulin --forward --features "cranelift,singlepass,test-cranelift,test-no-traps" --release -t120 --out Xml -- --skip traps:: --skip spec::linking
+          ls
```
Comment (nlewycky) [inline .github/workflows/coverage.yaml:24]: What's the `ls` for?

Comment (MarkMcCaskey) [reply .github/workflows/coverage.yaml:24]: old debug code

File: .github/workflows/coverage.yaml (hunk 1 above)
Comment (MarkMcCaskey) [reply .github/workflows/coverage.yaml:6]: Both, it doesn't hurt to have it, it's not needed but it's better to have it enabled in CI

Comment (nlewycky) [reply .github/workflows/coverage.yaml:6]: Sounds good!

File: tests/lib/wast/src/wast.rs [hunk 3]
```diff
@@ -171,6 +173,8 @@ impl Wast {
                 let result = self.perform_invoke(call);
                 self.assert_trap(result, message)?;
             }
+            #[cfg(feature = "test-no-traps")]
```
Comment (nlewycky) [inline tests/lib/wast/src/wast.rs:176]: Is there an issue filed either on us or on tarpaulin for this? Could you put a link to it in a comment here?

**Pull request merged**